From ae96aa945600d2e0a34b475c6717f90ecfca2cb1 Mon Sep 17 00:00:00 2001
From: Li Jun <b47624@freescale.com>
Date: Sat, 17 May 2014 03:15:34 +0800
Subject: [PATCH 1326/1594] ENGR00319720-2 usb: common: otg-fsm: start HNP
 polling timer in host state

commit e19d0282ed6e2587187ee6329c33332b1e320672 from
git://git.freescale.com/imx/linux-2.6-imx.git

This patch starts HNP polling timer when otg is set to be a_host
or b_host.

Signed-off-by: Li Jun <b47624@freescale.com>
---
 drivers/usb/common/usb-otg-fsm.c |    4 +++-
 include/linux/usb/otg-fsm.h      |    1 +
 2 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 61d538a..61dbf3b 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -1,7 +1,7 @@
 /*
  * OTG Finite State Machine from OTG spec
  *
- * Copyright (C) 2007,2008 Freescale Semiconductor, Inc.
+ * Copyright (C) 2007-2015 Freescale Semiconductor, Inc.
  *
  * Author:	Li Yang <LeoLi@freescale.com>
  *		Jerry Huang <Chang-Ming.Huang@freescale.com>
@@ -169,6 +169,7 @@ static int otg_set_state(struct otg_fsm *fsm, enum usb_otg_state new_state)
 		otg_set_protocol(fsm, PROTO_HOST);
 		usb_bus_start_enum(fsm->otg->host,
 				fsm->otg->host->otg_port);
+		otg_add_timer(fsm, HNP_POLLING);
 		break;
 	case OTG_STATE_A_IDLE:
 		otg_drv_vbus(fsm, 0);
@@ -203,6 +204,7 @@ static int otg_set_state(struct otg_fsm *fsm, enum usb_otg_state new_state)
 		 */
 		if (!fsm->a_bus_req || fsm->a_suspend_req_inf)
 			otg_add_timer(fsm, A_WAIT_ENUM);
+		otg_add_timer(fsm, HNP_POLLING);
 		break;
 	case OTG_STATE_A_SUSPEND:
 		otg_drv_vbus(fsm, 1);
diff --git a/include/linux/usb/otg-fsm.h b/include/linux/usb/otg-fsm.h
index a789463..733e21b 100644
--- a/include/linux/usb/otg-fsm.h
+++ b/include/linux/usb/otg-fsm.h
@@ -56,6 +56,7 @@ enum otg_fsm_timer {
 	B_DATA_PLS,
 	B_SSEND_SRP,
 	A_DP_END,
+	HNP_POLLING,
 
 	NUM_OTG_FSM_TIMERS,
 };
-- 
1.7.5.4

